### PR TITLE
Update to use CircleCI 2.0 instead of 1.0

### PR DESCRIPTION
**Pull request by arrouse**
This adds a `config.yml` file that has the same CI mechanics as the old `circle.yml` file.

This also updates Fastlane and Cocoapods to their latest version.

There also seem to be a few updates that others have made that didn't regenerate the template so the updated template is checked in as well.

### CODE REVIEW

File: PRODUCTNAME/app/fastlane/Pluginfile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Autogenerated by fastlane
+#
+# Ensure this file is checked in to source control!
+
+gem 'fastlane-plugin-install_provisioning_profiles', git: "https://github.com/arrouse/install_provisioning_profiles"
```
Comment (KingOfBrian) [inline PRODUCTNAME/app/fastlane/Pluginfile]: What's this? The URL Looks wrong

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      FL_OUTPUT_DIR: /Users/distiller/project/output
+      FASTLANE_LANE: beta
+    shell: /bin/bash --login -eo pipefail
+    working_directory: /Users/distiller/project
+    steps:
+      - checkout
+      - run: bundle install
+      - run: cd app && bundle exec fastlane $FASTLANE_LANE
+      - store_artifacts:
+          path: /Users/distiller/project/output/{{ cookiecutter.project_name }}.ipa
```
Comment (KingOfBrian) [inline {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: Is there anyway to get rid of some of the duplication above? Seems painful to maintain.

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# .circleci/config.yml
+version: 2
+jobs:
+  test:
+    macos:
+      xcode: "9.2.0"
+    environment:
+      FL_OUTPUT_DIR: /Users/distiller/project/output
+      FASTLANE_LANE: test
```
Comment (KingOfBrian) [inline {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: Not sure breaking out the lane name helps readability here.

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      FL_OUTPUT_DIR: /Users/distiller/project/output
+      FASTLANE_LANE: develop
+    shell: /bin/bash --login -eo pipefail
+    working_directory: /Users/distiller/project
+    steps:
+      - checkout
+      - run: bundle install
+      - run: cd app && bundle exec fastlane $FASTLANE_LANE
+      - store_artifacts:
+          path: /Users/distiller/project/output/{{ cookiecutter.project_name }}.ipa
```
Comment (KingOfBrian) [inline {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: The artifacts currently stores the full build output and the `junit.xml` file. Would be good to get those in the artifact directory as well

File: PRODUCTNAME/app/fastlane/Pluginfile (hunk 1 above)
Comment (arrouse) [reply PRODUCTNAME/app/fastlane/Pluginfile]: Currently we need to use a fork of this plugin until these small changes get merged in.  After that we won't have to specify a git url. https://github.com/dgyesbreghs/install_provisioning_profiles/pull/3

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml (hunk 2 above)
Comment (arrouse) [reply {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: I agree.  Ill update to use the FL_OUTPUT_DIR to reduce some of the duplicate path information, but I haven't found a way to do it yet without writing it in a script which I can go either way on, and Circle doesn't do it in any of their samples.

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml (hunk 3 above)
Comment (arrouse) [reply {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: Agreed.  Will remove.

File: {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml (hunk 4 above)
Comment (arrouse) [reply {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: Good point.  Ill look into it.

Comment (heyltsjay) [reply {{ cookiecutter.project_name | replace(' ', '') }}/.circleci/config.yml]: Isn't this managed by the fastfile currently? If either of you has a second to clarify what this means I can get all these changes in. @arrouse / @KingOfBrian

File: PRODUCTNAME/app/fastlane/Pluginfile (hunk 1 above)
Comment (ZevEisenberg) [reply PRODUCTNAME/app/fastlane/Pluginfile]: It looks like we should be able to point this back at `master` now.

**Pull request merged**